src/cmake-tools.ts

```diff
@@ -543,6 +530,49 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
           log.debug(localize('needs.reconfigure', 'The project needs to be reconfigured so that the changes saved in CMakeLists.txt have effect.'));
         }
       }
+
+      // For multi-root, the "onDidSaveTextDocument" will be received once for each project folder.
+      // To avoid misleading telemetry, consider the notification only for the active folder.
+      // There is always one active folder in a workspace and never more than one.
+      if (isActiveFolder(this.folder)) {
+        // "outside" evaluates whether the modified cmake file belongs to the active folder.
+        // Currently, we don't differentiate between outside active folder but inside any of the other
+        // workspace folders versus outside any folder referenced by the current workspace.
+        let outside: boolean = true;
+        let fileType: string | undefined;
+        if (str.endsWith("CMakeLists.txt")) {
```

Comment:
For path string comparisons, should we be doing case insensitive?

Reply:
Good idea.